Simple Inventory Management System/Simple Inventory Management System/Inventory .cs

```
@@ -0,0 +1,126 @@
+﻿
+namespace Simple_Inventory_Management_System
+{
+    public class Inventory
+    {
+        public List<Product> Products { get; private set; }
+        public Inventory(List<Product> products)
+        {
+            this.Products = products;
+        }
+        public Inventory()
+        {
+            this.Products = new List<Product>();
+        }
+        public bool Add(Product product)
+        {
+            if (IsNotValid(product))
+                return false;
+
+            var existingProduct = CheckIfExist(product);
+
+            if (existingProduct != null)
+            {
+                existingProduct.Price = product.Price;
+                IncreaseQuantity(existingProduct, product.Quantity);
+            }
+            else
+            {
+                Products.Add(product);
+            }
+
+            return true;
+        }
+        public void DisplayProducts()
+        {
+            if (!Products.Any()) 
+            {
+                Console.WriteLine("There are no products yet");
+                return;
+            }
+            foreach (Product product in Products) 
+            {
+                Console.WriteLine(product);
+            }
+        }
+        public bool Edit(Product product ) 
+        {
+            if (IsNotValid(product))
+            { return false; }
+
+            var existingProduct = CheckIfExist(product);
+            if (existingProduct == null)
+                return false;
```

as above, we should throw an exception that the product doesn't exist, and the caller to handle the exception.